cernopendata_client/downloader.py

```diff
@@ -28,6 +30,32 @@
 
 from .validator import validate_range
 from .printer import display_message
+from .verifier import get_file_checksum
+from .config import DOWNLOAD_RETRY_SLEEP_TIME
+
+
+def check_error(path=None, file_location=None, protocol=None, retry=None):
+    """Check downloaded file for error page."""
+    file_name = file_location.split("/")[-1]
+    file_path = path + "/" + file_name
+    error_page = (
+        os.path.getsize(file_path) == 3846
+        and get_file_checksum(file_path) == "adler32:a82d5324"
+    )
+    if error_page:
+        for _retry in range(0, retry + 1):
+            if _retry == retry:
+                display_message(msg_type="error", msg="Number of retries exceeded.")
+                sys.exit(1)
+            display_message(
+                msg_type="info", msg="Retrying {}/{}".format(_retry + 1, retry)
+            )
+            time.sleep(DOWNLOAD_RETRY_SLEEP_TIME)
+            download_single_file(
+                path=path, file_location=file_location, protocol=protocol
+            )
+            if not error_page:
```

Comment:
You are missing rechecking the newly downloaded file here, the `error_page` is not set inside the for loop, so even if the file is well downloaded, the loop does not exit.